src/Swift.Runtime/src/Swift/Data.cs

```diff
@@ -47,24 +49,25 @@ static TypeMetadata ISwiftObject.GetTypeMetadata()
 
     [UnmanagedCallConv(CallConvs = [typeof(CallConvSwift)])]
     [DllImport(KnownLibraries.SwiftFoundation, EntryPoint = "$s10Foundation4DataVMa")]
-    public static unsafe extern TypeMetadata PInvoke_getMetadata();
+    public static extern TypeMetadata PInvoke_getMetadata();
 
-    static unsafe ISwiftObject ISwiftObject.NewFromPayload(SwiftHandle handle)
+    static ISwiftObject ISwiftObject.NewFromPayload(IntPtr handle)
     {
         return new Data(handle);
     }
 
-    IntPtr ISwiftObject.MarshalToSwift(IntPtr swiftDest)
+    void ISwiftObject.MarshalToSwift(Span<byte> swiftDestSpan)
     {
         var metadata = SwiftObjectHelper<Data>.GetTypeMetadata();
```

Comment:
Why don't we check `IsValid` here?

Reply:
Added this check before adding to the cache:
```csharp
public static TypeMetadata GetTypeMetadata()
    {
        return TypeMetadata.Cache.GetOrAdd(typeof(T), _ =>
        {
            TypeMetadata metadata = T.GetTypeMetadata();
            if (!metadata.IsValid)
            {
                throw new InvalidOperationException($"Failed to retrieve type metadata for {typeof(T)}");
            }

            return metadata;
        });
    }
```